docs/src/api/public.md

```diff
@@ -0,0 +1,142 @@
+# Public API Reference
+
+## Modeling
+
+```@autodocs
+Modules = [PowerSystems]
+Pages   = ["PowerSystems.jl",
+           "branches.jl",
+           "components.jl",
+           "injection.jl",
+           "devices.jl",
+           "loads.jl",
+           "supplemental_constructors",
+           "generation.jl",
+           "storage.jl",
+           "services.jl",
+           "topological_elements.jl",
+           "dynamic_models.jl",
+           "static_models.jl",
+           "dynamic_models.jl",
+           "operational_cost.jl",
+           "definitions.jl"]
+Public = true
+Private = false
+```
+
+## TimeSeries
+
+```@autodocs
+Modules = [InfrastructureSystems]
+Pages   = ["abstract_time_series.jl",
+           "deterministic.jl",
+           "probabilistic.jl",
+           "scenarios.jl",
+           "single_time_series.jl",
+           "forecasts.jl"]
+Order = [:type, :function]
+Filter = t -> t ∉ [InfrastructureSystems.get_internal,
+                   InfrastructureSystems.set_internal!]
+```
+
+```@autodocs
+Modules = [InfrastructureSystems]
+Pages   = ["time_series_cache.jl"]
+Order = [:type, :function]
+Filter = t -> t ∈ [InfrastructureSystems.get_next_time_series_array!,
+                   InfrastructureSystems.get_next_time]
+```
+
+## System
+
+```@autodocs
+Modules = [PowerSystems]
+Pages   = ["base.jl"]
+Public = true
+Private = false
+Filter = t -> t ∈ [System]
+```
+
+```@autodocs
+Modules = [PowerSystems]
+Pages = ["parsers/power_system_table_data.jl",
+         "parsers/power_models_data.jl",
+         "parsers/TAMU_data.jl",
+         "parsers/psse_dynamic_data.jl"]
+Public = true
+Private = false
+Filter = t -> t ∈ [System]
+```
+
+```@autodocs
+Modules = [PowerSystems]
+Pages   = ["base.jl"]
+Public = true
+Private = false
+Filter = t -> t ∉ [System]
+```
+
+## Additional Component Methods
+
+```@autodocs
+Modules = [PowerSystems]
+Pages   = ["supplemental_accessors.jl"]
+Public = true
+Private = false
+```
+
+```@autodocs
+Modules = [InfrastructureSystems]
+Pages   = ["component.jl", "components.jl"]
+Filter = t -> t ∈ [InfrastructureSystems.get_time_series,
```

Comment:
ForecastCache, StaticTimeSeriesCache, TimeSeriesCache, get_next_time_series_array, get_next_time

Reply:
Those are at the top already in the TimeSeries section

Reply:
Perhaps I misunderstood.  I thought that this section would define the entire public API (including docstrings).  There would of course be some redundancy in topic-specific sections.

Reply:
It is just the API, all I did was organize it by theme and apply some filters to avoid duplicates.